packages/suite/src/utils/wallet/transactionUtils.ts

```diff
@@ -380,36 +380,48 @@ const getBitcoinRbfParams = (
         const addr = allAddresses.find(a => input.addresses?.includes(a.address));
         if (!addr) return []; // skip utxo, TODO: set some error? is it even possible?
         // re-create utxo from the input
-        return [
-            {
-                amount: input.value!,
-                txid: input.txid!,
-                vout: input.vout || 0,
-                address: addr!.address,
-                path: addr!.path,
-                blockHeight: 0,
-                confirmations: 0,
-            },
-        ];
+        return {
+            amount: input.value!,
+            txid: input.txid!,
+            vout: input.vout || 0,
+            address: addr!.address,
+            path: addr!.path,
+            blockHeight: 0,
+            confirmations: 0,
+            required: true,
+        };
     });
     // find change address and output
     let changeAddress: AccountAddress | undefined;
     const outputs: RbfTransactionParams['outputs'] = [];
     vout.forEach(output => {
-        const changeOutput = changeAddresses.find(a => output.addresses?.includes(a.address));
-        outputs.push({
-            type: changeOutput ? 'change' : 'payment',
-            address: output.addresses![0],
-            amount: output.value!,
-            formattedAmount: formatNetworkAmount(output.value!, account.symbol),
-        });
-        if (changeOutput) {
-            changeAddress = changeOutput;
+        if (!output.isAddress) {
+            // TODO: this should be done in trezor-connect, blockchain-link or even blockbook
+            // blockbook sends output.hex as scriptPubKey with additional prefix where: 6a - OP_RETURN and XX - data len
```

Comment:
What does `6a - OP_RETURN and XX - data len` mean? I don't get it. 🤷

Reply:
https://en.bitcoin.it/wiki/Script#Constants

blockbook returns hex data prefixed with 2 constants and we want data without it.

example:
https://tbtc1.trezor.io/api/v2/tx/6039e108e1dfc532c3ae9eceaf6f7e87e967083ea23ad2d7444383ce1ab50eca

data returned by BB in output.hex `6a03666f6f`
data retuned by BB in output.addresses![0] field: `OP_RETURN (foo)`
we want: `666f6f` (ascii foo)

unwanted prefix:
6a - OP_RETURN constant
03 - data length

Reply:
Aha! Thanks for the link to bitcoin wiki. I finally learned what the BTC scripts are. 🙈 

So if I understand this correctly `output.hex` and `output.addresses[0]` always hold the same data only differently encoded. Here we *only* use later. That might be what confused me. Why are we mentioning the `hex` field when it's not used in the code?

Maybe we could change the comment like so:
```
             // TODO: this should be done in trezor-connect, blockchain-link or even blockbook
             // Extract data from BlockBook format which takes form of `OP_RETURN (<data in ascii>)`.
```

I.e. completely omit the part about `output.hex`?

However, this is rather minor thing. So feel free to RBY. .)

Reply:
`Why are we mentioning the hex field when it's not used in the code?`

because `hex` should be used instead of `address[0]` which is totally confusing.
valid solution is to use `@trezor/utxo-lib` like [in connect](https://github.com/trezor/connect/blob/develop/src/js/core/methods/tx/refTx.js#L115) but is not available here and adding it as a dependency only for this operation (minor use case) is unnecessary.

that is why i've commented that it should be fixed on totally different level (connect/blockchain-link)

i'll update the comment as you suggested